File: pinyougou-sellergoods-interface/src/main/java/com/pinyougou/sellergoods/service/BrandService.java
package com.pinyougou.sellergoods.service;

import com.pinyougou.entity.PageResult;
import com.pinyougou.pojo.TbBrand;

import java.util.List;
import java.util.Map;

public interface BrandService {
    List<TbBrand> findAll();
    PageResult findPage(Integer pageNum,Integer pageSize);
    void insert(TbBrand brand);
    void update(TbBrand brand);
    TbBrand findOne(Long id);
    void delete(Long[] id);
    PageResult findPage(TbBrand brand, Integer pageNum, Integer pageSize);
    List<Map> selectOptionList();
}

File: pinyougou-manager-web/src/main/webapp/js/service/brandService.js
app.service("brandService",function ($http) {
    this.findAllBrands = function () {
        return $http.get("../findAllBrands.do");
    }

    this.findPage = function (page, rows) {
        return $http.get("../findPage.do?page="+page+"&rows="+rows);
    }

    this.save = function (entity) {
        return $http.post("../save.do",entity);
    }

    this.update = function (entity) {
        return $http.post("../update.do",entity);
    }

    this.findOne = function (id) {
        return $http.get("../findOne.do?id="+id);
    }

    this.delete = function (ids) {
        return  $http.get("../delete.do?ids="+ids);
    }

    this.search = function (page,rows,searchEntity) {
        return $http.post("../search.do?page="+page+"&rows="+rows,searchEntity);
    }

    this.selectOptionList = function () {
        return $http.get("../selectOptionList.do");
    }
});